src/libbson/examples/bson-streaming-reader.c

```diff
@@ -14,6 +14,12 @@
  * limitations under the License.
  */
 
+#ifdef __FreeBSD__
+#ifndef __BSD_VISIBLE
+#define __BSD_VISIBLE 1
```

Comment:
Suggest defining `__BSD_VISIBLE` in cmake following the [pattern here](https://github.com/mongodb/mongo-c-driver/blob/95d7a187e3b410793916d854806a81d6bff9a8a4/CMakeLists.txt#L265-L279).

That may remove the need for defining `__BSD_VISIBLE` in individual files. Since FreeBSD is not tested in Evergreen, I expect future code may also need `__BSD_VISIBLE` defined.